lib/api/api.dart

```diff
@@ -449,6 +453,16 @@ class ApiSingleton {
         List<dynamic> jsonAnswer = json.decode(response.body);
         var list = <Report>[];
         for (var item in jsonAnswer) {
+          double lat, lon;
+          if (item['location_choice'] == 'current') {
+              lat = item['current_location_lat'] ?? Utils.defaultLocation.latitude;
+              lon = item['current_location_lon'] ?? Utils.defaultLocation.longitude;
+          } else {
+              lat = item['selected_location_lat'] ?? Utils.defaultLocation.latitude;
+              lon = item['selected_location_lon'] ?? Utils.defaultLocation.longitude;
+          }
+          var cityName = await getCityNameFromCoords(lat, lon);
+          item['display_city'] = cityName;
```

Comment:
I suggest relocating this code within the Report constructor as it seems more fitting there. This logic appears unrelated to the communication with the API.

Reply:
I agree, I tried first to do it first in the Report.fromJson constructor but got:
```
The modifier 'async' can't be applied to the body of a constructor.
Try removing the modifier.dart[invalid_modifier_on_constructor](https://dart.dev/diagnostics/invalid_modifier_on_constructor)
```

Reply:
Have you tried using a separate async function inside the class and call it?

```dart
class Report {
 // properties here

  Report() {
   // current code here
    _initialize();
  }

  Future<void> _initialize() async {
   // code to get appropiate lat and lon here
    var cityName = await getCityNameFromCoords(lat, lon);
    displayCity = cityName;
  }
}
```

Reply:
I just tried it and it works, nice. I agree that's better than it being in the api.dart